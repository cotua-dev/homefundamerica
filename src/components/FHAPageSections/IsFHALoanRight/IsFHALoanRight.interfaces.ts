export interface FetchIsFHALoanRightImageQuery {
    isFHALoanRightImage: {
        publicURL: string;
    };
}

export interface Reason {
    key: number;
    reason: string;
}
