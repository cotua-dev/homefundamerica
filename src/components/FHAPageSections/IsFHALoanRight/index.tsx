import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import {
    FetchIsFHALoanRightImageQuery,
    Reason,
} from './IsFHALoanRight.interfaces';
import { isFHALoanRightSection } from './IsFHALoanRight.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const IsFHALoanRight: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchIsFHALoanRightImageQuery>(graphql`
        query {
            isFHALoanRightImage: file(name: {eq: "AdobeStock_338493617"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    const listOfReasons: Reason[] = [
        {
            key: 0,
            reason: 'To take advantage of flexible qualifications and credit requirements',
        },
        {
            key: 1,
            reason: 'When your down payment of home equity is less than 20% of the loan amount',
        },
        {
            key: 2,
            reason: 'To reduce monthly mortgage payments by refinancing an existing loan',
        },
        {
            key: 3,
            reason: 'To access home equity through a cash-out refinance option',
        },
    ];

    return (
        <section
            id={isFHALoanRightSection}
            className={sharedStyles.loanNegative}
        >
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.isFHALoanRightImage.publicURL}
                    alt="Is FHA loan right image"
                />
            </div>
            <div className={sharedStyles.contentWrapper}>
                <h2>{`Is an FHA Loan Right For You?`}</h2>
                <strong>{`Situations in which an FHA loan makes sense:`}</strong>
                <ul>
                    {listOfReasons.map((reason: Reason) => (
                        <li key={reason.key}>{reason.reason}</li>
                    ))}
                </ul>
                <button
                    className={sharedStyles.applyButton}
                    type="button"
                >Apply Now</button>
            </div>
        </section>
    );
}
