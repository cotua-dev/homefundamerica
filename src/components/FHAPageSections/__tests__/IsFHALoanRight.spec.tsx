import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { IsFHALoanRight } from '../IsFHALoanRight';

describe('IsFHALoanRight Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<IsFHALoanRight/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
