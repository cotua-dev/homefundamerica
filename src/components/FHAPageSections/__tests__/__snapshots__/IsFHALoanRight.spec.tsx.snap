// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsFHALoanRight Component renders 1`] = `
<section
  id="isFHALoanRightSection"
>
  <div>
    <img
      alt="Is FHA loan right image"
    />
  </div>
  <div>
    <h2>
      Is an FHA Loan Right For You?
    </h2>
    <strong>
      Situations in which an FHA loan makes sense:
    </strong>
    <ul>
      <li>
        To take advantage of flexible qualifications and credit requirements
      </li>
      <li>
        When your down payment of home equity is less than 20% of the loan amount
      </li>
      <li>
        To reduce monthly mortgage payments by refinancing an existing loan
      </li>
      <li>
        To access home equity through a cash-out refinance option
      </li>
    </ul>
    <button
      type="button"
    >
      Apply Now
    </button>
  </div>
</section>
`;
